api/v1/databaseengine_types.go

```diff
@@ -73,6 +81,32 @@ type DatabaseEngineList struct {
 	Items           []DatabaseEngine `json:"items"`
 }
 
+// Versions struct represents available versions of database engine components.
+type Versions struct {
+	Engine map[string]*Component            `json:"engine,omitempty"`
+	Backup map[string]*Component            `json:"backup,omitempty"`
+	Proxy  map[string]map[string]*Component `json:"proxy,omitempty"`
+	Tools  map[string]map[string]*Component `json:"tools,omitempty"`
+}
+
+// Component contains information of the database engine component.
+// Database Engine component can be database engine, database proxy or tools image path
+type Component struct {
+	Critical  bool   `json:"critical,omitempty"`
+	ImageHash string `json:"imageHash,omitempty"`
+	ImagePath string `json:"imagePath,omitempty"`
+	Status    string `json:"status,omitempty"`
+}
+
+func (d DatabaseEngine) RecommendedBackupImage() string {
+	for _, component := range d.Status.AvailableVersions.Backup {
+		if component.Status == "recommended" {
+			return component.ImagePath
+		}
+	}
+	return ""
```

Comment:
This is dangerous, PXC has 2 recommended backup versions, one for 8.0 and another one for 5.7

Reply:
It is. However we use this only for PSMDB. I can make the integration with DatabaseEngine tighter in the database cluster controller, however, this function used only for PSMDB

Reply:
I see you only used this for PSMDB... But I see someone shooting itself in the foot with this in the future...

Reply:
It won't because we will refactor it in the future with a new CR definition and use a version service to get images

Reply:
And that's in the very near future so no one will forget that this function cannot be used :crossed_fingers:

Reply:
It should be refactored and some new functions should be added depending on the version of the operator/engine that should be used for proxy/backup/engine images. This is just an idea to demonstrate the way how we can add this integration